src/wovnio/wovnphp/Headers.php

```diff
@@ -392,6 +401,17 @@ public function removeLang($uri, $lang = null)
 
         $lang_code = $this->store->convertToCustomLangCode($lang);
         $default_lang = $this->store->settings['default_lang'];
+        $url_pattern_name = $this->store->settings['url_pattern_name'];
+        $custom_domain_langs = $this->store->settings['custom_domain_langs'];
+        if ($url_pattern_name == 'custom_domain' && isset($custom_domain_langs)) {
+            $current_lang_domain = array_search($lang_code, $custom_domain_langs);
+            $default_lang_domain = array_search($default_lang, $custom_domain_langs);
+            $current_lang_path = parse_url('http://' . $current_lang_domain . '/')['path'];
+            $default_lang_path = parse_url('http://' . $default_lang_domain . '/')['path'];
+            return str_replace($current_lang_path, $default_lang_path, $uri);
```

Comment:
To generate original HTML path, path needs to be generated from custom_domain_langs.
But custom_domain_langs's url includes host.
So, host part should be removed first.